spec/rspec/rails/extensions/active_support/json_spec.rb

```diff
@@ -0,0 +1,196 @@
+        expect(subject.send(patch)).to be(:stubbed)
+      end
+
+      it "captures a set expectation" do
+        expect(subject).to receive(patch)
+        subject.send(patch)
+      end
+
+      include_examples "verifies the patch arguments" if opts[:verified]
+    end
+  end
+
+  context "using a `spy`" do
+    subject(:a_spy) { spy }
+
+    include_examples "intercepts Active Support's patch", :to_json
+    include_examples "intercepts Active Support's patch", :as_json
+  end
+
+  context "using an `instance_spy`" do
+    subject(:the_spy) { instance_spy(String) }
+
+    include_examples "intercepts Active Support's patch", :to_json, :verified => true
+    include_examples "intercepts Active Support's patch", :as_json, :verified => true
+  end
+
+  context "using a `class_spy`" do
+    subject(:the_spy) { class_spy(String) }
+
+    include_examples "intercepts Active Support's patch", :to_json, :verified => true
+    include_examples "intercepts Active Support's patch", :as_json, :verified => true
+  end
+
+  context "using an `object_spy`" do
+    subject(:the_spy) { object_spy('test') }
+
+    include_examples "intercepts Active Support's patch", :to_json
+    include_examples "intercepts Active Support's patch", :as_json
+  end
+
+  context "using a `double`" do
+    subject(:a_double) { double }
+
+    include_examples "ignores Active Support's patch", :to_json
+    include_examples "ignores Active Support's patch", :as_json
+  end
+
+  context "using an `instance_double`" do
+    subject(:the_double) { instance_double(String) }
+
+    include_examples "ignores Active Support's patch", :to_json, :verified => true
+    include_examples "ignores Active Support's patch", :as_json, :verified => true
+  end
+
+  context "using a `class_double`" do
+    subject(:the_double) { class_double(String) }
+
+    include_examples "ignores Active Support's patch", :to_json, :verified => true
+    include_examples "ignores Active Support's patch", :as_json, :verified => true
+  end
+
+  context "using an `object_double`" do
+    subject(:the_double) { object_double('test') }
+
+    include_examples "ignores Active Support's patch", :to_json
+    include_examples "ignores Active Support's patch", :as_json
+  end
+
+  it "doesn't interfere with a partial mock's native representation" do
+    obj = { :foo => "a test", :bar => 123 }
+    allow(obj).to receive(:equal?).and_return(false)
+    expect(obj.to_json).to eq("{\"foo\":\"a test\",\"bar\":123}")
+    expect(obj.as_json).to eq({ "foo" => "a test", "bar" => 123 })
+  end
+
+  context "with BasicObject" do
+    if defined?(::BasicObject)
+      before(:example) do
+        basic_object = Class.new do
+          undef :as_json
+          undef :to_json
+        end
+        stub_const("BasicObject", basic_object)
+      end
+    else
+      # Sanity check to ensure Rails didn't change this on us
+      before(:context) do
+        expect(BasicObject).not_to respond_to(:as_json)
```

Comment:
https://travis-ci.org/rspec/rspec-rails/jobs/120927983#L543

Unless I'm mistaken, BasicObject will never exist here because we defined checked it above. Do we know why the undefined constant is happening here?